compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -373,6 +382,9 @@ impl<Tag: Provenance, Extra> Allocation<Tag, Extra> {
 
     /// Reads a *non-ZST* scalar.
     ///
+    /// If `read_pointer` is `true`, this will also read provenance; otherwise (if the machine
```

Comment:
Maybe name all of these `read_pointer` args to `read_provenance` or `preserve_provenance`? It wasn't immediately obvious to me what `read_pointer` means.